### PR TITLE
Use trusted publisher and environment

### PR DESCRIPTION
**Pull request by eccles**
Setup PyPi for datatrails-python to use a trusted publisher. Delete any API tokens in PyPi for datatrails-python.

In github repo delete PYPI secrets and create an environmant called release that is restricted to the main branch.

Testing: can only be tested by merging and releasing: if this fails it will be refined and remerged and rereleased. Any wheels so uploaded will be marked as alpha and yanked to prevent access by third parties.

### CODE REVIEW

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -37,21 +45,29 @@ jobs:
...
+         python3 -m build --sdist
+         python3 -m build --wheel
+      shell: bash
+
+    - name: Publish to PyPi
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        verbose: true
+        attestations: true
+        # skip-existing: true
```
Comment (landintrees) [inline .github/workflows/package.yml:63]: Is this intended to be commented out?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -1,5 +1,9 @@
-# This workflow will upload a Python Package using Twine when a release is created
-# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+# This workflow will upload a Python Package using a release environmet and a trusted publisher.
```
Comment (waarg) [inline .github/workflows/package.yml:1]: typo - "environmet"

File: .github/workflows/package.yml [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
+# See PyPi management in the datatrails confluence service for an explanation.
```
Comment (waarg) [inline .github/workflows/package.yml:2]: You could add the link to the confluence page in this comment?

File: .github/workflows/package.yml [hunk 4]
```diff
@@ -37,21 +45,28 @@ jobs:
...
+
+    - name: Create wheel
       run: |
         rm -f archivist/about.py
         ./scripts/version.sh
-        python3 -m build --sdist
-        python3 -m build --wheel
-        twine check dist/*
-        twine upload dist/*
+         python3 -m build --sdist
```
Comment (waarg) [inline .github/workflows/package.yml:54]: is there an extra indentation here?

**Review by waarg — APPROVED** [review]

**Pull request merged**